code/controllers/garbage.dm

```diff
@@ -88,35 +91,36 @@ var/soft_dels = 0
 	dels_count++
 
 /*
- * NEVER USE THIS FOR ANYTHING OTHER THAN /atom/movable
- * OTHER TYPES CANNOT BE QDEL'D BECAUSE THEIR LOC IS LOCKED OR THEY DON'T HAVE ONE.
+ * NEVER USE THIS FOR /atom OTHER THAN /atom/movable
+ * BASE ATOMS CANNOT BE QDEL'D BECAUSE THEIR LOC IS LOCKED.
  */
-/proc/qdel(const/atom/movable/AM, ignore_pooling = 0)
+/proc/qdel(const/datum/D, ignore_pooling = 0, ignore_destroy = 0)
 	//writepanic("[__FILE__].[__LINE__] (no type)([usr ? usr.ckey : ""])  \\/proc/qdel() called tick#: [world.time]")
-	if(isnull(AM))
+	if(isnull(D))
 		return
 
 	if(isnull(garbageCollector))
-		del(AM)
+		del(D)
 		return
 
-	if(!istype(AM))
-		WARNING("qdel() passed object of type [AM.type]. qdel() can only handle /atom/movable types.")
-		del(AM)
+	if(istype(D, /atom) && !istype(D, /atom/movable))
+		WARNING("qdel() passed object of type [D.type]. qdel() cannot handle unmovable atoms.")
+		del(D)
 		garbageCollector.hard_dels++
 		garbageCollector.dels_count++
 		return
 
 	//We are object pooling this.
-	if(("[AM.type]" in masterPool) && !ignore_pooling)
-		returnToPool(AM)
+	if(("[D.type]" in masterPool) && !ignore_pooling)
```

Comment:
You have to merge the two types of pooling (or at the very least the pooling lists) otherwise this wont work for datums